crypto/dh/dh_check.c

```diff
@@ -119,7 +119,14 @@ int DH_check(const DH *dh, int *ret)
     if (t2 == NULL)
         goto err;
 
-    if (dh->q) {
+    if (dh->q != NULL) {
+        if (BN_ucmp(dh->p, dh->q) > 0)
```

Comment:
is negative q possible?

Reply:
Yes, but nothing terribly bad happens.
I've just checked that with my original reproducer and the Primality Check fails
immediatly here:
https://github.com/openssl/openssl/blob/eec805ee71356c06f9a86192fa06507c3bb92b09/crypto/bn/bn_prime.c#L165
becasuse a is negative.